problem_builder/mentoring.py

```diff
@@ -776,3 +793,98 @@ def workbench_scenarios():
         Scenarios displayed by the workbench. Load them from external (private) repository
         """
         return loader.load_scenarios_from_path('templates/xml')
+
+
+class MentoringWithExplicitStepsBlock(BaseMentoringBlock, StudioContainerWithNestedXBlocksMixin):
+    """
+    An XBlock providing mentoring capabilities with explicit steps
+    """
+    # Settings
+    display_name = String(
+        display_name=_("Title (Display name)"),
+        help=_("Title to display"),
+        default=_("Mentoring Questions (with explicit steps)"),
+        scope=Scope.settings
+    )
+
+    editable_fields = ('display_name',)
+
+    @lazy
+    def questions(self):
+        """ Get the usage_ids of all of this XBlock's children that are "Questions" """
+        return list(chain.from_iterable(self.runtime.get_block(step_id).steps for step_id in self.steps))
```

Comment:
@itsjeyd nit: this is rather expensive operation - it queries mongo for each individual block. I'm not 100% sure there is a caching mechanism in place (probably it is) or if any "collection" methods are exposed to Xblocks, but at least `self.runtime.get_block(step_id) for step_id in self.steps` could be "cached" in a lazy property.